dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -68,25 +81,28 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations }: any) =>
     }
   }, [userConfigs?.dataConfig, visualizations.vis.name]);
 
-  const updateList = (value: string, index: number, name: string, field: string) => {
-    const list = { ...configList };
-    let listItem = { ...list[name][index] };
-    listItem = {
-      ...listItem,
-      [field === 'custom_label' ? 'alias' : field]: value,
-    };
-    if (field === 'label') {
-      listItem.name = value;
+  const updateList = (value: string, field: string) => {
+    if (value !== '') {
+      const { index, name } = selectedConfigItem;
+      const list = { ...configList };
```

Comment:
I do not see you updating this `configList` or `list` variable in this function. Hence I think, it is unnecessary to create another variable just to hold the values of this variable by using a spread operation. Let me know, if I am missing something.